### PR TITLE
Tags for Boxes, shared with Beneficiaries

### PR DESCRIPTION
**Pull request by vahidbazzaz**
This PR contains the changes to support unified tags for the Beneficiaries and Boxes.

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[Tags for Boxes, shared with Beneficiaries #large](https://trello.com/c/XjNwO3sL)

File: include/people_edit.php [hunk 1]
```diff
@@ -4,58 +4,78 @@
     $action = 'people_edit';
 
     if ($_POST) {
-        // delete a transaction of a person
-        if ('delete' == $_POST['do']) {
-            $ids = explode(',', $_POST['ids']);
-            list($success, $message, $redirect) = listDelete('transactions', $ids);
+        [$message, $id] = db_transaction(function () use ($table, $settings) {
```
Comment (aerinsol) [inline include/people_edit.php:7]: confused. How does this relate to tags?

File: include/people_edit.php [hunk 2]
```diff
@@ -78,9 +98,9 @@
         FROM 
             people
         LEFT JOIN
-            people_tags ON people_tags.people_id = people.id
+            tags_relations ON tags_relations.object_id = people.id AND tags_relations.object_type = "People"
         LEFT JOIN
-            tags ON tags.id = people_tags.tag_id 
+            tags ON tags.id = tags_relations.tag_id AND tags_relations.object_type = "People" AND tags.deleted IS NULL
```
Comment (aerinsol) [inline include/people_edit.php:103]: When did we add a deleted field to tags?

File: include/tags.php [hunk 3]
```diff
@@ -43,10 +48,79 @@
...
+                                        tags.deleted IS NULL
+                                            AND tags.camp_id = :campId
+                                            AND tags.id = :tagId
+                                    GROUP BY tags.id", ['campId' => $_SESSION['camp']['id'], 'tagId' => $id]);
+
+                    $originType = $result['origin_type'];
+                    $boxesCount = intval($result['boxes_count']);
+                    $beneficiariesCount = intval($result['beneficiaries_count']);
+
+                    define('BOX_WARNING', 'Changing the apply to will remove those tags once it has been saved, as there are about '.$boxesCount.' boxes that applied tags');
```
Comment (aerinsol) [inline include/tags.php:99]: Box warning: 'WARNING: Changing the tag type will remove '.$boxesCount.' existing tags from boxes."

File: include/people_edit.php (hunk 2 above)
Comment (vahidbazzaz) [reply include/people_edit.php:103]: @aerinsol this is removal tags that sometime ago we added https://trello.com/c/no6uIJAr

File: include/tags.php [hunk 4]
```diff
@@ -43,10 +48,79 @@
...
+                    define('PEOPLE_WARINING', 'Changing the apply to will remove those tags once it has been saved, as there are about '.$beneficiariesCount.' beneficiaries that applied tags');
```
Comment (aerinsol) [inline include/tags.php:100]: `define ('PEOPLE_WARNING'` is misspelled. 

Also, recommended warning text: 'WARNING: Changing the tag type will remove '.$beneficiariesCount.' existing tags that have been applied to beneficiaries."

**Review by aerinsol — CHANGES_REQUESTED** [review]
Nicely done! Just a few small change requests on my side. Also, I think I noticed that for the badge displays in the "Edit Boxes" views, the tags seem to appear in reverse order to what they appear like in the "Manage Boxes" view. Can you check that? @vahidbazzaz

File: include/people_edit.php (hunk 1 above)
Comment (vahidbazzaz) [reply include/people_edit.php:7]: @aerinsol Just wrapping the transactions over the entire queries while editing beneficiaries (including the tags applied)

**Review by aerinsol — APPROVED** [review]

File: include/stock_edit.php [hunk 5]
```diff
@@ -44,20 +67,48 @@
...
+                GROUP_CONCAT(tags.color ORDER BY tags.seq) AS tagcolors
             FROM stock AS s 
-            LEFT OUTER JOIN products AS p ON p.id = s.product_id 
-            LEFT OUTER JOIN genders AS g ON g.id = p.gender_id 
-            LEFT OUTER JOIN locations AS l ON l.id = s.location_id 
+                LEFT OUTER JOIN products AS p ON p.id = s.product_id 
+                LEFT OUTER JOIN genders AS g ON g.id = p.gender_id 
+                LEFT OUTER JOIN locations AS l ON l.id = s.location_id 
+                LEFT JOIN tags_relations ON tags_relations.object_id = stock.id AND tags_relations.object_type = "Stock"
+                LEFT JOIN tags ON tags.id = tags_relations.tag_id
```
Comment (HaGuesto) [inline include/stock_edit.php:81]: tags.deleted

File: include/stock_edit.php [hunk 6]
```diff
@@ -66,7 +114,8 @@
     $cmsmain->assign('include', 'cms_form.tpl');
     addfield('hidden', '', 'id');
     // put a title above the form
-    $cmsmain->assign('title', 'Box');
+    $cmsmain->assign('peopletitle', 'Box ');
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply include/stock_edit.php:120]: we renamed it but it works generally

**Review by HaGuesto — APPROVED** [review]

**Pull request merged**